private-templates-service/server/src/db/providers/StorageProvider.ts

```diff
@@ -0,0 +1,175 @@
+// Base class for all the StorageProviders.
```

Comment:
You should add a readme about the StorageProvider and how it is expected to be used